### PR TITLE
Run nancy validation for all dependencies

### PR DESCRIPTION
**Pull request by SVilgelm**

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -15,6 +15,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
```
Comment (sayboras) [inline .github/workflows/pr.yml:18]: should I be in another job as this job is focusing on golangci-lint ? I think golangci-lint and nancy are two independent tasks, they can be running in parallel as well :)

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -15,6 +15,11 @@ jobs:
...
+      # We cannot use nancy-github-action because it is outdated, so it's better to use the latest
+      # docker image for the validation
+      - name: nancy
+        run: go list -m all | docker run -i sonatypecommunity/nancy:latest
```
Comment (sayboras) [inline .github/workflows/pr.yml:22]: We might need to test it out for negative case to make sure it failed here. I faced a few cases github action didn't fail as expected, so just a precautious step.

Comment (SVilgelm) [reply .github/workflows/pr.yml:22]: I'm using it in several projects, an at work and it's failing :)

Comment (SVilgelm) [reply .github/workflows/pr.yml:22]: recently I had to disable it on a weekend because the server was unavailable

File: .github/workflows/pr.yml (hunk 1 above)
Comment (SVilgelm) [reply .github/workflows/pr.yml:18]: ok, let me move it

File: .github/workflows/pr.yml (hunk 2 above)
Comment (sayboras) [reply .github/workflows/pr.yml:22]: oh I see, we can just let it run as non-required job for sometime first, then change settings later once we see fit.

Comment (SVilgelm) [reply .github/workflows/pr.yml:22]: then I need to move it to a separate file

Comment (SVilgelm) [reply .github/workflows/pr.yml:22]: like `extra`

Comment (SVilgelm) [reply .github/workflows/pr.yml]: done

**Comment by SVilgelm** [comment]
> /lgtm

then approve? ;-)

File: .github/workflows/pr-extra.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  vulns:
+    name: Vulnerability scanner
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+      # We cannot use nancy-github-action because it is outdated, so it's better to use the latest
+      # docker image for the validation
+      - name: nancy
+        run: go list -m all | docker run -i sonatypecommunity/nancy:latest
```
Comment (iwankgb) [inline .github/workflows/pr-extra.yml:19]: You should rather use:
```
go list -json -m all | nancy
```
(see [deprecation notice](https://github.com/sonatype-nexus-community/nancy#what-is-the-best-usage-of-nancy))

File: .github/workflows/pr.yml (hunk 2 above)
Comment (sayboras) [reply .github/workflows/pr.yml]: oh actually, you can still have it in the same pr workflow, mandatory setting is per `job`. 

Anyway, no harm to keep it in another file.

File: .github/workflows/pr-extra.yml (hunk 3 above)
Comment (SVilgelm) [reply .github/workflows/pr-extra.yml:19]: thanks I didn't know

Comment (SVilgelm) [reply .github/workflows/pr-extra.yml]: done
